spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -1352,6 +1352,15 @@ object CometSparkSessionExtensions extends Logging {
     org.apache.spark.SPARK_VERSION >= "4.0"
   }
 
+  def isComplexTypeReaderEnabled(conf: SQLConf): Boolean = {
```

Comment:
I find the naming confusing here. This method determines if we are using `native_datafusion` or `native_iceberg_compat` (which both use DataFusion's `ParquetExec`). This is no logic related to complex types.

Complex type support was a big motivation for adding these new scans, but it doesn't seem to make sense to refer to complex types in the changes in this PR.

This is just a nit, and we can rename the methods in a future PR.